### PR TITLE
upgrade clean up

### PR DESCRIPTION
**Pull request by JFranklin3000**
##### In this PR
-fixes upgrade drop on drones and characters that can transform
-simplified the error message  

##### Testing
-Make sure that drops on Drones and actors that can transform work And other characters do not.

### CODE REVIEW

File: lang/en.json [hunk 1]
```diff
@@ -371,11 +371,10 @@
     "Upgrade": "Upgrade",
     "UpgradeAvailability": "Availability",
     "UpgradeBenefit": "Benefit",
-    "UpgradeDroneError": "Drones can only accept drone upgrades",
+    "UpgradeDropError": "This upgrade can not be dropped here",
```
Comment (phildominguez) [inline lang/en.json:374]:
```suggestion
    "UpgradeDropError": "Upgrade of type {upgrade_type} cannot be dropped on {actor_type}",
```

File: module/sheets/actor-sheet.mjs [hunk 2]
```diff
@@ -503,11 +503,12 @@ export class Essence20ActorSheet extends ActorSheet {
...
-        ui.notifications.error(game.i18n.format(game.i18n.localize('E20.UpgradeDroneError')));
-        return false;
-      } else if (this.actor.type == 'transformer' && sourceItem.system.type != 'armor') {
-        ui.notifications.error(game.i18n.format(game.i18n.localize('E20.UpgradeTransformerError')));
+      if (this.actor.type == 'companion' && this.actor.system.type == 'drone' && sourceItem.system.type == 'drone') {
+        super._onDropItem(event, data);
+      } else if (this.actor.system.canTransform && sourceItem.system.type == 'armor') {
+        super._onDropItem(event, data);
+      } else {
+        ui.notifications.error(game.i18n.format(game.i18n.localize('E20.UpgradeDropError')));
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:511]:
```suggestion
      const actor_type = this.actor.type;
      const upgrade_type = sourceItem.system.type;

      if (actor_type == 'companion' && this.actor.system.type == 'drone' && upgrade_type == 'drone') {
        super._onDropItem(event, data);
      } else if (this.actor.system.canTransform && upgrade_type == 'armor') {
        super._onDropItem(event, data);
      } else {
        ui.notifications.error(game.i18n.format('E20.UpgradeDropError', {upgrade_type, actor_type}));
```
Untested, but something like this should work for a better error message

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheets/actor-sheet.mjs [hunk 3]
```diff
@@ -493,6 +493,8 @@ export class Essence20ActorSheet extends ActorSheet {
 
     const sourceItem = await fromUuid(data.uuid);
     if (!sourceItem) return false;
+    const actor_type = this.actor.type;
+    const upgrade_type = sourceItem.system.type;
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:497]: I would put these in the upgrade case since they're only used there

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheets/actor-sheet.mjs (hunk 3 above)
Comment (JFranklin3000) [reply module/sheets/actor-sheet.mjs]: rolled back changes

**Review by phildominguez — APPROVED** [review]

**Pull request merged**